packages/parser/lib/interfaces/ILiteralCreator.ts

```diff
@@ -0,0 +1,7 @@
+import QuoteTypes from "../types/QuoteTypes";
+
+export default interface ILiteralCreator {
```

Comment:
what's the difference between `interface` and `type` in ts. i know i tend to use `type` and the differences are subtle, but just curious because i don't know when to use one of the other.

also, would it  be possible to have the creator interfaces and the actual type (here: `ILiteralCreator` and `Literal`) in the same ts file since the types are so similar, it'd be nice to only have to update one file instead of 2

Reply:
For example, here the `interface` is been used as a contract in the case, send me anything that you want I don't care as long as has these things and the `type` is using for shaping things and giving them names, in the form of this object with these properties it's called `Literal`, let me know if that makes sense.

I think the ILiteralCreator is going to be a more generic thing in the future that why is in another file, if that doesn't happen it can be moved for sure.